collect_app/src/main/java/org/odk/collect/android/utilities/FileUtils.java

```diff
@@ -261,113 +257,37 @@ private static String actualCopy(File sourceFile, File destFile) {
         }
     }
 
-    public static HashMap<String, String> parseXML(File xmlFile) {
-        final HashMap<String, String> fields = new HashMap<>();
-        final InputStream is;
-        try {
-            is = new FileInputStream(xmlFile);
-        } catch (FileNotFoundException e1) {
-            Timber.d(e1);
-            throw new IllegalStateException(e1);
-        }
-
-        InputStreamReader isr;
-        try {
-            isr = new InputStreamReader(is, "UTF-8");
-        } catch (UnsupportedEncodingException uee) {
-            Timber.w(uee, "Trying default encoding as UTF 8 encoding unavailable");
-            isr = new InputStreamReader(is);
-        }
-
-        final Document doc;
-        try {
-            doc = XFormParser.getXMLDocument(isr);
-        } catch (IOException e) {
-            Timber.e(e, "Unable to parse XML document %s", xmlFile.getAbsolutePath());
-            throw new IllegalStateException("Unable to parse XML document", e);
-        } finally {
-            try {
-                isr.close();
-            } catch (IOException e) {
-                Timber.w("%s error closing from reader", xmlFile.getAbsolutePath());
-            }
-        }
-
-        final String xforms = "http://www.w3.org/2002/xforms";
-        final String html = doc.getRootElement().getNamespace();
-
-        final Element head = doc.getRootElement().getElement(html, "head");
-        final Element title = head.getElement(html, "title");
-        if (title != null) {
-            fields.put(TITLE, XFormParser.getXMLText(title, true));
-        }
+    /**
+     * Given a form definition file, return a map containing form metadata. The form ID is required
+     * by the specification and will always be included. Title and version are optionally included.
+     * If the form definition contains a submission block, any or all of submission URI, base 64 RSA
+     * public key, auto-delete and auto-send may be included.
+     */
+    public static HashMap<String, String> getMetadataFromFormDefinition(File formDefinitionXml) {
```

Comment:
I feel like this might be a nice opportunity to move the meta data representation to a data class rather than a Map.  What do you think?

Reply:
I agree. This is part of what I ended up backing out of. I considered using the existing `Form` class which is billed as a data transfer object but not used very much. I think that could be appropriate but I'd like to discuss where the code to go from an XML form definition and/or JavaRosa FormDef to one of those should go (e.g. https://github.com/opendatakit/collect/commit/decf3925e7fa28791f5524b66aaad5f252e860fa#diff-824505d34be688f35cf8665de92654e3R245). Some other kind of adapter? Once I started going down that path, I quickly ran into weird inconsistencies between `DiskSyncTask` and `FormDownloader` which both call the current `getMetadataFromFormDefinition`. I think most of the work they do to validate the form properties is not needed but it's different and hard to disentangle. I'm also concerned about things like https://github.com/opendatakit/collect/commit/decf3925e7fa28791f5524b66aaad5f252e860fa#diff-0ceefebf9b2ba619c3af7866f81294fbR285 where it's not entirely clear whether the media path needs to be separately saved (`DiskSyncTask` doesn't do this). I'm worried that we'd end up going back and forth a bunch in review and QA. If you think it's important to do now, let's talk through the high-level strategy before I dive further.

Reply:
To be clear, there's some urgency around this because it's going to get used for some behavior needed in #3447. On top of that, deeper changes would likely conflict with other open PRs. So what I might suggest is that we go forward with this for now, discuss the refactor strategy, and then I can start playing with it with the goal of getting something mergeable in v1.26. What do you think?

Reply:
Ah cool. I'm happy with an issue then and we can stick it in 1.26 👍. That lets us have a proper dig into it.

Reply:
Filed issue at #3461.